src/main/java/liquibase/ext/opensearch/changelog/OpenSearchHistoryService.java

```diff
@@ -113,7 +114,7 @@ protected List<RanChangeSet> queryRanChangeSets() throws DatabaseException {
                     .search(s -> s.index(this.getDatabaseChangeLogTableName()), RanChangeSet.class);
             return response.hits().hits().stream()
                     .map(Hit::source)
-                    .collect(Collectors.toList());
+                    .collect(Collectors.toList()); // do not use toList directly as it returns an immutable list!
```

Comment:
I think this comment is not accurate. Collectors.toList() does return immutable list as well (AFAIK). Moreover, from the javadocs (https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/stream/Collectors.html#toList()):

> Returns a Collector that accumulates the input elements into a new List. There are no guarantees on the type, mutability, serializability, or thread-safety of the List returned; if more control over the returned List is required, use [toCollection(Supplier)](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/stream/Collectors.html#toCollection(java.util.function.Supplier)).

Reply:
good point! sorry, i just tested this with my JDK without reading the javadoc and `Stream#toList` returned a mutable list 😅